controller/scheduler/scheduler.go

```diff
@@ -256,11 +256,11 @@ func (s *Scheduler) Run() error {
 		return err
 	}
 
-	var err error
-	s.isLeader, err = s.discoverd.Register()
+	isLeader, err := s.discoverd.Register()
 	if err != nil {
 		return err
 	}
+	s.HandleLeaderChange(isLeader)
```

Comment:
This has the side effect of calling [rectifyAll](https://github.com/jzila/flynn/blob/scheduler-v2/controller/scheduler/scheduler.go#L521), should we not just leave it as it was and log a message here?